state/data/reducer.js

```diff
@@ -0,0 +1,19 @@
+import * as types from './types';
+
+const initialState = {
+  movies: {}
+};
+
+const data = (state = initialState, action) => {
+  switch (action.type) {
+    case types.MERGE_MOVIES_LIST:
+      return {
+        ...state,
+        movies: action.movies
```

Comment:
This action overrides movies in state. If you want to merge you need to spread old movies with new:
```
return {
  ...state,
  movies: {
    ...state.movies,
    ...action.movies
  }
}
```